pkgs/java_http/lib/src/java_client.dart

```diff
@@ -63,16 +63,19 @@ class JavaClient extends BaseClient {
     final responseHeaders = await events.next as Map<String, String>;
 
     Stream<List<int>> responseBodyStream(Stream<dynamic> events) async* {
-      await for (final event in events) {
-        if (event is ClientException) {
-          throw event;
-        } else if (event is List<int>) {
-          if (event.length == 1 && event[0] == -1) {
-            receivePort.close();
+      try {
+        await for (final event in events) {
+          if (event is List<int>) {
+            yield event;
+          } else if (event is ClientException) {
+            throw event;
+          } else if (event == null) {
             return;
           }
-          yield event;
         }
+      } finally {
+        // TODO: Should we kill the isolate here?
```

Comment:
Won't it exit on it's own?

Reply:
Hey @brianquinlan,

I think it does exit on its own normally although I'm not 100% sure about when the user cancels the `StreamSubscription` of the `StreamedResponse.stream`. But perhaps we don't have to worry too much about this?